PWG/DevNanoAOD/AliAnalysisNanoAODCuts.h

```diff
@@ -223,6 +223,65 @@ class AliAnalysisNanoAODEventCuts : public AliAnalysisCuts
   ClassDef(AliAnalysisNanoAODEventCuts, 3); // event cut object for nano AOD filtering
 };
 
+class AliAnalysisNanoAODMCParticleCuts : public AliAnalysisCuts
+{
+public:
+  AliAnalysisNanoAODMCParticleCuts();
+  virtual ~AliAnalysisNanoAODMCParticleCuts() {}
+  virtual bool IsSelected(TObject* obj);  // TObject should be an AliAODMCParticle
+  virtual bool IsSelected(TList* /* list */) {
+    return kTRUE;
+  }
+  bool GetSelectPrimaries() const {
+    return fDoSelectPrimaries;
+  }
+  void SetSelectPrimaries(bool doIt) {
+    fDoSelectPrimaries = doIt;
+  }
+  bool GetSelectChargedParticles() const {
+    return fDoSelectCharged;
+  }
+  void SetSelectChargedParticles(bool doIt) {
+    fDoSelectCharged = doIt;
+  }
+  float GetMinPt() const {
+    return fMinPt;
+  }
+  void SetMinPt(float var) {
+    fMinPt = var;
+  }
+  float GetMaxEta() const {
+    return fMaxEta;
+  }
+  void SetMaxEta(float var) {
+    fMaxEta = var;
+  }
+  std::vector<int> &GetKeepParticles() {
+    return fPDGToKeep;
+  }
+  void SetKeepParticles(std::vector<int> pdgCodes) {
+    fPDGToKeep = pdgCodes;
+    fForceKeepParticles = true;
+  }
```

Comment:
In addition, let's add a function AddKeepParticles(int pdgCode) which adds it to the list